lib/alephant/broker/response/base.rb

```diff
@@ -48,10 +52,19 @@ def add_no_cache_headers
           log
         end
 
+        def add_etag_allow_header
+          headers.merge!("Access-Control-Expose-Headers" => "ETag")
+        end
+
         def component_not_modified(headers, request_env)
-          return false if headers["Last-Modified"].nil? && headers["ETag"].nil?
+          return false if request_env.if_modified_since.nil? && request_env.if_none_match.nil?
+
+          !request_env.if_modified_since.nil? && headers["Last-Modified"] == request_env.if_modified_since ||
+            !request_env.if_none_match.nil? && unquote_etag(headers["ETag"]) == unquote_etag(request_env.if_none_match)
+        end
```

Comment:
This last statement, although perfectly correct is getting a little unreadable...  Maybe separate out into variables?

```ruby
last_modified_match = !request_env.if_modified_since.nil? && headers["Last-Modified"] == request_env.if_modified_since
etag_match = !request_env.if_none_match.nil? && unquote_etag(headers["ETag"]) == unquote_etag(request_env.if_none_match)

last_modified_match || etag_match
```

or maybe even helper functions?